### PR TITLE
Add `hatchling` recipe

### PR DESCRIPTION
**Pull request by ofek**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ofek** [comment]
@conda-forge/help-python this recipe is ready for review. I don't understand the error because I see the package here https://anaconda.org/conda-forge/pathspec

File: recipes/hatchling/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - python
+  run:
+    - editables >=0.2  # [py>3],==0.*
+    - importlib-metadata  # [py<38]
+    - packaging >=20.9  # [py<3],==20.*
+    - pathspec >=0.9,==0.*
+    - pluggy >=0.13  # [py<3],==0.*
+    - python
+    - toml >=0.10.2  # [py<3],==0.10.*
+    - tomli >=2.0.0  # [py>3],==2.0.*
```
Comment (henryiii) [inline recipes/hatchling/meta.yaml:30]:
```suggestion
    - editables >=0.2
    - importlib-metadata
    - packaging >=20.9
    - pathspec >=0.9
    - pluggy >=0.13
    - python
    - tomli >=2.0
```

File: recipes/hatchling/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/hatchling-{{ version }}.tar.gz
+  sha256: 33816429b78d0f069717ccce53a2ec722c79412c86928caa2e01a93b4eaf31a5
+
+build:
```
Comment (henryiii) [inline recipes/hatchling/meta.yaml:12]:
```suggestion
build:
  noarch: python
```

File: recipes/hatchling/meta.yaml (hunk 1 above)
Comment (henryiii) [reply recipes/hatchling/meta.yaml:30]: Conda-forge doesn't support Python 2 anymore, and you can't set Python requirements on a pure Python package (which this should be, `noarch: python`), and [no upper caps](https://iscinumpy.dev/post/bound-version-constraints/), please. They can be injected later if something breaks.

File: recipes/hatchling/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  entry_points:
+    - hatchling = hatchling.cli:hatchling
+
+requirements:
+  host:
+    - pip
+    - python >=2.7
```
Comment (henryiii) [inline recipes/hatchling/meta.yaml:22]: If toml is required for python < 3.6, I think I'd just use >3.6 here. Conda-forge doesn't build for anything less than 3.7 anymore anyway.

File: recipes/hatchling/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  run:
+    - editables >=0.2
+    - importlib-metadata
+    - packaging >=20.9
+    - pathspec >=0.9
```
Comment (henryiii) [inline recipes/hatchling/meta.yaml:27]: This (as well as anything else needed for bootstrapping) needs to also be in `host`.

Comment (henryiii) [reply recipes/hatchling/meta.yaml:27]: (This meaning pathspec)

**Comment by ofek** [comment]
@conda-forge/help-python this recipe is ready for review; with the help of @henryiii the CI is now passing 😄

**Review by dopplershift — APPROVED** [review]

**Pull request merged**